doc/evp_skey.md

```diff
@@ -0,0 +1,99 @@
+EVP_SKEY Design
+===============
+
+We currently presume that symmetric keys are represnted as octet buffer. It may
+be not relevant for the situation of hardware-backed providers, such as
+PKCS#11 or TPM.
```

Comment:
For hardware-backed providers, wouldn't it be reasonable to use the `load` sort of functionality, which `keymgmt` already provides?